projects/plugins/jetpack/_inc/client/state/initial-state/reducer.js

```diff
@@ -395,16 +396,30 @@ export const getUpgradeUrl = ( state, source, userId = '', planDuration = false
 		source += '-monthly';
 	}
 
-	return (
-		'https://jetpack.com/redirect/?' +
-		`source=${ source }&site=${ getSiteRawUrl( state ) }` +
-		( affiliateCode ? `&aff=${ affiliateCode }` : '' ) +
-		( uid ? `&u=${ uid }` : '' ) +
-		( subsidiaryId ? `&subsidiaryId=${ subsidiaryId }` : '' ) +
-		( isCurrentUserLinked( state ) ? '' : '&unlinked=1' ) +
-		( purchaseToken ? `&purchasetoken=${ purchaseToken }` : '' ) +
-		( calypsoEnv ? `&calypso_env=${ calypsoEnv }` : '' )
-	);
+	const redirectArgs = {
+		site: getSiteRawUrl( state ),
+	};
+
+	if ( affiliateCode ) {
+		redirectArgs.aff = affiliateCode;
+	}
+	if ( uid ) {
+		redirectArgs.u = uid;
+	}
+	if ( subsidiaryId ) {
+		redirectArgs.subsidiaryId = subsidiaryId;
+	}
+	if ( ! isCurrentUserLinked( state ) ) {
+		redirectArgs.query = 'unlinked=1';
+	}
+	if ( purchaseToken ) {
+		redirectArgs.purchasetoken = purchaseToken;
```

Comment:
Is `purchasetoken` a separate argument, or we should put that into `query` as well?
On the "Plans" link it does go into the `query`: https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/_inc/client/components/navigation/index.jsx#L60

Reply:
Yes, I highlighted it in the p2 and was waiting for the discussion [here](p9dueE-36H-p2#comment-5499) to decide. I'll update the PR with the decision.

Reply:
done